Feature/client-service

Antonio-LAndrade-Babel opened the pull request:
Se crea el servicio para el Client. Se compone de:

Clases del paquete «dtos»
- ClientDto

Clases del paquete «service/client»:
- ClientService
- ClientServiceImpl

Clases del paquete «repository»:
- ClientRepository

Antonio-LAndrade-Babel committed:
Creación del Servicio Producto

Creación del Servicio Producto, incluyendo su repositorio.

Antonio-LAndrade-Babel committed:
Se cambia Stock de primitiva a clase envolvente

Antonio-LAndrade-Babel committed:
Se cambia price de clase primitiva a clase envolvente.

Antonio-LAndrade-Babel committed:
Implementación de ProductServiceImpl

Antonio-LAndrade-Babel committed:
Implementación del StoreService

Se añaden las clases de servicio:
- StoreService
- StoreServiceImpl

Y la clase de repositorio:
- StoreReporitory

Antonio-LAndrade-Babel committed:
Implementación del ClientService

Se crea el servicio para el Client. Se compone de:

Clases del paquete «service»:
- ClientService
- ClientServiceImpl

Clases del paquete «repository»:
- ClientRepository

RubenDelReyBabel commented on src/main/java/com/babelgroup/service/ClientService.java:
```diff
@@ -0,0 +1,14 @@
+
+public interface ClientService {
+    Client createClient(Client client);
+    Client updateClient(String clientId, Client clientDetails);
```
No es necesario pasar el id del cliente, este ya está en el objeto

RubenDelReyBabel commented on src/main/java/com/babelgroup/service/ProductService.java:
```diff
@@ -0,0 +1,12 @@
+
+public interface ProductService {
+    Product createProduct(Product product);
+    Product updateProduct(String id, Product productDetails);
```
No es necesario el id del producto, ya está en el objeto producto

Antonio-LAndrade-Babel committed:
Actualización de client-service para trabajar con DTO

- Se añade ClientDto
- Modificado el servicio para trabajar directamente con el DTO en lugar de con el objeto per se.

Antonio-LAndrade-Babel committed:
Se elimina ClientService del paquete raíz service
